include/swift/AST/CASTBridging.h

```diff
@@ -126,178 +170,151 @@ extern "C" {
+                BridgedSourceLoc cThrowsLoc, BridgedSourceLoc cParamLLoc,
+                BridgedArrayRef params, BridgedSourceLoc cParamRLoc,
+                void *_Nullable returnType, BridgedDeclContext cDeclContext);
 void FuncDecl_setBody(void *fn, void *body);
 
-void *SimpleIdentTypeRepr_create(void *ctx, void *loc, BridgedIdentifier id);
+void *SimpleIdentTypeRepr_create(BridgedASTContext cContext, BridgedSourceLoc cLoc, BridgedIdentifier id);
 
-void *UnresolvedDotExpr_create(void *ctx, void *base, void *dotLoc,
-                               BridgedIdentifier name, void *nameLoc);
+void *UnresolvedDotExpr_create(BridgedASTContext cContext, void *base, BridgedSourceLoc cDotLoc,
+                               BridgedIdentifier name, BridgedSourceLoc cNameLoc);
 
-void *ClosureExpr_create(void *ctx, void *body, void *dc);
+void *ClosureExpr_create(BridgedASTContext cContext, void *body, BridgedDeclContext cDeclContext);
 
 void NominalTypeDecl_setMembers(void *decl, BridgedArrayRef members);
 
-struct DeclContextAndDecl {
-  void *declContext;
-  void *nominalDecl;
-  void *decl;
-};
-
-struct DeclContextAndDecl
-StructDecl_create(void *ctx, void *loc, BridgedIdentifier name, void *nameLoc,
-                  void *_Nullable genericParams, void *dc);
-struct DeclContextAndDecl ClassDecl_create(void *ctx, void *loc,
+BridgedDeclContextAndDecl
+StructDecl_create(BridgedASTContext cContext, BridgedSourceLoc cLoc, BridgedIdentifier name, BridgedSourceLoc cNameLoc,
+                  void *_Nullable genericParams, BridgedDeclContext cDeclContext);
+BridgedDeclContextAndDecl ClassDecl_create(BridgedASTContext cContext, BridgedSourceLoc cLoc,
                                            BridgedIdentifier name,
-                                           void *nameLoc, void *dc);
-
-void *ArrayTypeRepr_create(void *ctx, void *base, void *lsquareLoc,
-                           void *rsquareLoc);
-void *DictionaryTypeRepr_create(void *ctx, void *keyType, void *valueType,
-                                void *lsquareLoc, void *colonloc,
-                                void *rsquareLoc);
-void *OptionalTypeRepr_create(void *ctx, void *base, void *questionLoc);
-void *ImplicitlyUnwrappedOptionalTypeRepr_create(void *ctx, void *base,
-                                                 void *exclamationLoc);
-void *MetatypeTypeRepr_create(void *ctx, void *baseType, void *typeLoc);
-void *ProtocolTypeRepr_create(void *ctx, void *baseType, void *protoLoc);
-void *PackExpansionTypeRepr_create(void *ctx, void *base, void *repeatLoc);
-void *TupleTypeRepr_create(void *ctx, BridgedArrayRef elements, void *lParenLoc,
-                           void *rParenLoc);
-void *MemberTypeRepr_create(void *ctx, void *baseComponent,
+                                           BridgedSourceLoc cNameLoc, BridgedDeclContext cDeclContext);
+
+void *ArrayTypeRepr_create(BridgedASTContext cContext, void *base, BridgedSourceLoc cLSquareLoc,
```

Comment:
Just a thought: Theoretically the parameter names wouldn’t need to have a `c` in front of them in the header file. I assume you only need them in the `.cpp` file to distinguish the boxed C type from the unwrapped/casted C++ type. 

Happy to keep it as-is though to have the header file consistent with the implementation file.

Reply:
Yeah, that's true. I couldn't decide to be honest so I just left it consistent. I don't really mind either way though.

Reply:
It’s probably easier to keep it consistent between the `.h` and `.cpp` files because then you can copy-paste the function declaration when creating a new function. Otherwise I’ll be chasing people to drop the `c` prefix in the  header  file for the next months.